system/mp/mpcomm.cpp

```diff
@@ -2237,6 +2237,37 @@ int CMPConnectThread::run()
                 connectHdr.id[0].get(_remoteep);
                 connectHdr.id[1].get(hostep);
 
+                // HPCC-28125 - if remoteep IP != peerEp IP then log warning - do we change remoteep ?
+                //              if hostep != listen ep then log warning - do we change hostep ?
+                // **VERY CONCERNED** about pods and local IPs and Network Address Translation
+
+                if (!_remoteep.ipequals(peerEp))
+                {
+                    StringBuffer rIp, pIp;
+                    _remoteep.getIpText(rIp);
+                    peerEp.getIpText(pIp);
+                    PROGLOG("MP: WARNING: remote client address (%s:%u) != peer address (%s:%u)", rIp.str(), _remoteep.port, pIp.str(), peerEp.port);
+                    _remoteep.set(peerEp);
+                    connectHdr.id[0].set(_remoteep);
+                    // should we sock->close() and continue ?
+                }
+
+                SocketEndpoint listenEp;
+                listensock->getEndpoint(listenEp);
+                unsigned short lPort = listenEp.port;
+                IpAddress myLocalIp;
+                GetHostIp(myLocalIp);
+                if (!hostep.ipequals(myLocalIp))
+                {
+                    StringBuffer hIp, lIp;
+                    hostep.getIpText(hIp);
+                    myLocalIp.getIpText(lIp);
+                    PROGLOG("MP: WARNING: remote client trying to reach (%s:%u) != listen address (%s:%u)", hIp.str(), hostep.port, lIp.str(), lPort);
+                    hostep.set(lPort, myLocalIp);
```

Comment:
see comment above, I'm not sure this (client notion of Dali IP being different from Dali's notion of its IP) actually causes any problems...
but I am wondering if changing this (to Dali's notion of itself) may cause problems for the clients that will no longer see a match in packets coming back.

Probably pretty obvious if it breaks things on a cloud system where they will mismatch..

Reply:
I agree, I am not sure changing hostep matters or causes problems.  Lets test and decide what is best.